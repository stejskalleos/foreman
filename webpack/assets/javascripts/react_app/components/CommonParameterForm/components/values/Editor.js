import React from 'react';
import PropTypes from 'prop-types';
import AceEditor from 'react-ace';

const Editor = ({ id, name, mode, value, onChange, fullScreen, isMasked }) => (
  <AceEditor
    mode={mode}
    id={id}
    name={name}
    theme="github"
    width="100%"
    height="150px"
    tabSize={2}
    editorProps={{ $blockScrolling: 'Infinity' }}
    value={`${value}`}
    onChange={val => onChange(val)}
    className={`common-param-editor ${fullScreen && 'fullscreen'} ${isMasked &&
      'mask-editor'}`}
    showPrintMargin={false}
    showLineNumbers
  />
);

Editor.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
  mode: PropTypes.string.isRequired,
  value: PropTypes.any,
  onChange: PropTypes.func.isRequired,
  fullScreen: PropTypes.bool,
  isMasked: PropTypes.bool,
};

Editor.defaultProps = {
  id: '',
  value: '',
  fullScreen: false,
  isMasked: false,
};

export default Editor;
