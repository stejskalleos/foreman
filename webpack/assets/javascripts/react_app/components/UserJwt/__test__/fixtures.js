import { noop } from '../../../common/helpers';
import { STATUS } from '../../../constants';

export const formProps = {
  handleSubmit: noop,
};

const token = 'token23';

export const fieldOk = {
  apiStatus: STATUS.RESOLVED,
  token,
};

export const fieldError = {
  apiStatus: STATUS.ERROR,
  token,
};

export const fieldPending = {
  apiStatus: STATUS.PENDING,
  token,
};
