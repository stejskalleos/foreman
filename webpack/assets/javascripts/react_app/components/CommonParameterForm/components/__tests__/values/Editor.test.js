import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import { editorFixtures } from '../fixtures';
import Editor from '../../values/Editor';


describe('Editor', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(Editor, editorFixtures));
});
