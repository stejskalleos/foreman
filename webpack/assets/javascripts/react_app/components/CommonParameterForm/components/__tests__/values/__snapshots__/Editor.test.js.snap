// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor rendering with fullScreen 1`] = `
<ReactAce
  className="common-param-editor fullscreen false"
  cursorStart={1}
  editorProps={
    Object {
      "$blockScrolling": "Infinity",
    }
  }
  enableBasicAutocompletion={false}
  enableLiveAutocompletion={false}
  focus={false}
  fontSize={12}
  height="150px"
  highlightActiveLine={true}
  id="common_parameter_value"
  maxLines={null}
  minLines={null}
  mode="text"
  name="common_parameter[value]"
  navigateToFileEnd={true}
  onChange={[Function]}
  onLoad={null}
  onPaste={null}
  onScroll={null}
  placeholder={null}
  readOnly={false}
  scrollMargin={
    Array [
      0,
      0,
      0,
      0,
    ]
  }
  setOptions={Object {}}
  showGutter={true}
  showLineNumbers={true}
  showPrintMargin={false}
  style={Object {}}
  tabSize={2}
  theme="github"
  value="something"
  width="100%"
  wrapEnabled={false}
/>
`;

exports[`Editor rendering without fullScreen 1`] = `
<ReactAce
  className="common-param-editor false false"
  cursorStart={1}
  editorProps={
    Object {
      "$blockScrolling": "Infinity",
    }
  }
  enableBasicAutocompletion={false}
  enableLiveAutocompletion={false}
  focus={false}
  fontSize={12}
  height="150px"
  highlightActiveLine={true}
  id="common_parameter_value"
  maxLines={null}
  minLines={null}
  mode="text"
  name="common_parameter[value]"
  navigateToFileEnd={true}
  onChange={[Function]}
  onLoad={null}
  onPaste={null}
  onScroll={null}
  placeholder={null}
  readOnly={false}
  scrollMargin={
    Array [
      0,
      0,
      0,
      0,
    ]
  }
  setOptions={Object {}}
  showGutter={true}
  showLineNumbers={true}
  showPrintMargin={false}
  style={Object {}}
  tabSize={2}
  theme="github"
  value="something"
  width="100%"
  wrapEnabled={false}
/>
`;
